MarkBind/markbind
# Warn users if PlantUML diagrams are found but Graphviz is not installed

**What is the purpose of this pull request?**




**Overview of changes:**
Related to #1903
- Add a check to verify that Graphviz is installed before processing PlantUML diagrams.
- This will be done once per build/refresh (in the case of markbind serve)
- A warning will be logged to the console
	- ![image](https://user-images.githubusercontent.com/41845017/163904240-eb71f2d0-638e-4608-bf3f-be12235687e7.png)
- I am leaving the decision and the work to fail hard (or not) to development in #1863, as that will involve changing existing test cases and how we decide what kinds of errors to totally stop the build. This PR simply logs the warning but does not error out.
- Include a way to disable the warning:
	- https://deploy-preview-1912--markbind-master.netlify.app/userguide/sitejsonfile#plantumlcheck
	- ![image](https://user-images.githubusercontent.com/41845017/166176946-f03aa13b-4a49-4f4e-95f3-1d6f96eb46dc.png)

- Lastly, make the warning in our UG even more explicit:
	- https://deploy-preview-1912--markbind-master.netlify.app/userguide/components/imagesanddiagrams#diagrams
	- ![image](https://user-images.githubusercontent.com/41845017/166176881-741cc1bb-6682-48fa-8fb6-87e875484792.png)



**Anything you'd like to highlight / discuss:**
nil

**Testing instructions:**
- Checkout to this PR
- insert the following (or any other puml code) in a .md file:
```md
<puml width="300">
@startuml
alice -> bob ++ : hello
bob -> bob ++ : self call
bob -> bib ++  #005500 : hello
bob -> george ** : create
return done
return rc
bob -> george !! : delete
return success
@enduml
</puml>
```
- Run `markbind build` or `markbind serve`, if Graphviz is installed, no warnings should be found, else, a warning should be found (exactly once even if u have multiple PlantUML diagrams)



**Proposed commit message: (wrap lines at 72 characters)**
Add Graphviz check when PlantUML diagrams are used

Without Graphviz, PlantUML diagrams generated may
contain errors. Users may not be aware that diagrams are
generated incorrectly if they do not serve up the pages.

Let's add a check and warn users in the console if the
installation of Graphviz is missing.

This helps to keep users informed in the console when
the needed installation is missing. Users are also informed
of a way to disable the warning in `site.json`.



---

**Checklist:** :ballot_box_with_check:
